### ISSUE TITLE
Change behavior of 'backend'

### ISSUE THREAD

**Issue by punyidea** [esa/torchquad#153]
# Feature
After upgrading from 0.2.3 to 0.3.0 I encountered an issue where the backend grid had changed from a pytorch tensor to a numpy array. After making a cursory read of the documentation, I attempted setting the `backend` flags both in the integrator and using `set_backend()`. After this did not work, I was about to file an issue when I looked more closely and found that this input was ignored when the `integration_bounds` variable was not a tensor. I was then able to fix my issue. However, I think this decision should be reconsidered, as I did not find it intuitive.

## Desired Behavior / Functionality
`backend` should be less ignored - at the very least, a warning should be raised stating that it has been ignored in favor of the integration bounds.

**Comment by gomezzz** [comment]
Hi @punyidea , thanks for the feedback!

Yes, with the new multi-framework support it is a bit tricky to decide, which backend should be given priority under which circumstances.

Could you maybe post a minimal code example what you did, what happened and what you would have expected?

We can change the priority of the different backend selections and add warnings very easily but I don't want to accidentally make it worse. :)

Thanks!

**Comment by punyidea** [comment]
Thanks for taking my feedback into consideration, I like the project overall!

Sure thing, here's my MWE.

    import torch
    import numpy as np
    import torchquad as tq
    
    def scaled_tensor(x):
        a = torch.tensor(2.)
        return torch.sum(a*x,axis=-1)
    def integrate_simpson(f,bounds):
        integrator = tq.Simpson()
        return integrator.integrate(f,N =10201,dim=2,integration_domain=bounds,backend='torch')

    x = integrate_simpson(scaled_tensor,bounds=np.array([[-1.,1.],[-1.,1.]]))
    print(type(x))

Intuitively to me, the type of the bounds has less important priority than the backend information, so I would expect (a) the output to be a tensor, and (b) the integration grid supplied to it internally to be a tensor. I find numpy arrays quite convenient ways of storing information.
I will note that it worked fine, as expected when `bounds` was a list of lists. Maybe a warning could be raised if the type of integration bounds is different from the `backend` variable? Or it could be converted if the types differ? The MWE above also applies when using `set_backend()`

Cheers!

**Comment by gomezzz** [comment]
Thanks for the example! Yea, I agree that this should at least give a warning and the manually specified backend should take precedence over the bounds inferred one. :)

@FHof do you want to have a look at this? Otherwise, I will take care of it :)

**Comment by FHof** [comment]
Yes, I also think there should be a warning if the backends don't match.
The backend is determined in the [`_setup_integration_domain`](https://github.com/esa/torchquad/blob/e0d82657d08492d4718be90a0212d69b2e61c6e8/torchquad/integration/utils.py#L116) function.

In addition to the backend, the function also determines the dtype.
If `integration_domain` is a list,
the backend for the integration is defined by the backend argument or set to a globally configured value (if backend is None),
and the dtype is set to a globally configured value.
On the other hand, if `integration_domain` is a tensor, both the backend and dtype are inferred from it.

I've created a PR for this issue: #154
Since it is currently possible to specify a dtype locally with `integration_domain` and this dtype may differ from a globally configured dtype, it may be unclear which dtype should be selected if a user passes a backend argument together with an `integration_domain` tensor, so I've added the warning.
